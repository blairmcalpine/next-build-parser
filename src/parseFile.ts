import type { PathOrFileDescriptor } from 'node:fs';
import fs from 'node:fs';
import { parse, type ParseOptions } from './parse.js';

export const parseFile = (
  file: PathOrFileDescriptor,
  options?: ParseOptions,
) => {
  const data = fs.readFileSync(file, 'utf8');
  return parse(data, options);
};
