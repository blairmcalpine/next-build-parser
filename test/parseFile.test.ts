import { expect, describe, it } from 'vitest';
import { parseFile } from '../src/parseFile.js';

describe('parseFile', () => {
  it('should parse the build file', () => {
    const result = parseFile('test/samples/basic.txt');
    expect(result).toMatchSnapshot();
  });
});
