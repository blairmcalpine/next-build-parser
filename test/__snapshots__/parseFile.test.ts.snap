// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseFile > should parse the build file 1`] = `
{
  "routes": [
    {
      "firstLoad": 125,
      "icon": "λ",
      "name": "/",
      "size": 1.1,
      "type": "Dynamic",
    },
    {
      "firstLoad": 86.2,
      "icon": "λ",
      "name": "/_not-found",
      "size": 0.85,
      "type": "Dynamic",
    },
    {
      "firstLoad": 0,
      "icon": "λ",
      "name": "/api/trpc/[trpc]",
      "size": 0,
      "type": "Dynamic",
    },
    {
      "firstLoad": 140,
      "icon": "λ",
      "name": "/create",
      "size": 15.5,
      "type": "Dynamic",
    },
    {
      "firstLoad": 125,
      "icon": "λ",
      "name": "/popular",
      "size": 1.1,
      "type": "Dynamic",
    },
    {
      "firstLoad": 132,
      "icon": "λ",
      "name": "/puzzle/[id]",
      "size": 8.15,
      "type": "Dynamic",
    },
    {
      "firstLoad": 0,
      "icon": "ℇ",
      "name": "/puzzle/[id]/opengraph-image",
      "size": 0,
      "type": "Edge",
    },
    {
      "firstLoad": 125,
      "icon": "λ",
      "name": "/recent",
      "size": 1.1,
      "type": "Dynamic",
    },
  ],
  "shared": [
    {
      "name": "chunks/472-9100b5fcfec8f88c.js",
      "size": 30,
    },
    {
      "name": "chunks/fd9d1056-853251486b1823c6.js",
      "size": 53.3,
    },
    {
      "name": "chunks/main-app-842b2bfec3a562ae.js",
      "size": 0.23,
    },
    {
      "name": "chunks/webpack-b7a1470890cdd00b.js",
      "size": 1.79,
    },
  ],
  "sharedTotal": 85.3,
  "unit": "kB",
}
`;
